spec/inputs/tcp_spec.rb

```diff
@@ -269,6 +269,39 @@ def get_port
     end
   end
 
+  it "should flush codec after client disconnects" do
+    # verifies fix for https://github.com/logstash-plugins/logstash-input-tcp/issues/90
+    conf = <<-CONFIG
+      input {
+        tcp {
+          port => #{port}
+          codec => multiline {
+              pattern => "^\s"
+              what => "previous"
+          }
+        }
+      }
+    CONFIG
+
+    data = "a\n 1\n 2\nb\n 1"
+    event_count = 2
+
+    events = input(conf) do |pipeline, queue|
+      socket = Stud::try(5.times) { TCPSocket.new("127.0.0.1", port) }
+      socket.puts(data)
+      socket.close
+
+      event_count.times.collect {
+        # use non-blocking pop with sleep so that a missing queue item doesn't cause the test to hang
+        # if a "queue empty" error occurs here, the codec was not properly flushed
+        sleep(2)
+        queue.pop(true)
+      }
+    end
```

Comment:
I am not sure I like the hard sleep here, since it will basically force that test to take 4 seconds and the 2 seconds is arbitrary here, could be too much but could be not enough depending, we don't know.
since we are only testing on the number of events maybe using a spin loop with a timeout like
```
Timeout.timeout(timeout) do
  sleep(0.1) until queue.size >= 2
end
```
WDYT?

Reply:
That sounds like it would work better. I wrapped the blocking calls to `queue.pop` with a timeout which should accomplish the same thing.